LandLord-core/src/main/java/biz/princeps/landlord/placeholderapi/LLExpansion.java

```diff
@@ -33,75 +49,104 @@ public String getVersion() {
         if (player == null) {
             return null;
         }
-        final int maxClaimPermission = pl.getPlayerManager().getMaxClaimPermission(player);
-        final int landCount = wg.getRegionCount(player.getUniqueId());
-        final IOwnedLand region = wg.getRegion(player.getLocation());
-
-        switch (s) {
-            case "ownedLands":
-                return String.valueOf(landCount);
-
-            case "claims":
-                final IPlayer iPlayer = pl.getPlayerManager().get(player.getUniqueId());
-                if (iPlayer == null) {
-                    pl.getLogger().warning("A placeholder is trying to load %landlord_claims% before async loading of the " +
-                            "player has finished! Use FinishedLoadingPlayerEvent!");
-                    return "NaN";
-                }
-                return String.valueOf(iPlayer.getClaims());
-
-            case "remainingClaims":
-                final IPlayer iPlayer2 = pl.getPlayerManager().get(player.getUniqueId());
-                if (iPlayer2 == null) {
-                    pl.getLogger().warning("A placeholder is trying to load %landlord_remainingClaims% before async loading of the " +
-                            "player has finished! Use FinishedLoadingPlayerEvent!");
-                    return "NaN";
-                }
-                return String.valueOf(iPlayer2.getClaims() - landCount);
 
-            case "currentLandOwner":
-                if (region != null && region.getOwner() != null) {
-                    return region.getOwnersString();
-                }
-                return "∅";
-
-            case "currentLandMembers":
-                final String members;
-                if (region != null && !(members = region.getMembersString()).isEmpty()) {
-                    return members;
-                }
-                return "∅";
-
-            case "currentLandName":
-                return wg.getLandName(player.getLocation().getChunk());
-
-            case "nextLandPrice":
-                return String.valueOf(pl.getCostManager().calculateCost(player.getUniqueId()));
-
-            case "currentLandRefund":
-                final int regionCount = wg.getRegionCount(player.getUniqueId());
-                return String.valueOf(pl.getCostManager().calculateCost(regionCount - 1) * pl.getConfig().getDouble(
-                        "Payback"));
-
-            case "maxLimitPermission":
-                return String.valueOf(maxClaimPermission);
-
-            case "remainingFreeLands":
-                final int freeLands = pl.getConfig().getInt("Freelands");
-
-                if (landCount <= freeLands) {
-                    return String.valueOf((Math.min(maxClaimPermission, freeLands)) - landCount);
-                } else {
-                    return "0";
+        try {
+            cache.get(player.getName() + "_" + placeholder, () -> {
+                switch (placeholder) {
```

Comment:
Its probably cleaner to move this whole cache calculation/switch statement to its own method.
This would at least remove two indentations.

Reply:
Right, will be moved in next commit!